Objects/longobject.c

```diff
@@ -2581,14 +2529,15 @@ long_from_non_binary_base(const char *start, const char *end, Py_ssize_t digits,
         /* carry off the current end? */
         if (c) {
             assert(c < PyLong_BASE);
-            if (Py_SIZE(z) < size_z) {
+            if (_PyLong_DigitCount(z) < size_z) {
                 *pz = (digit)c;
-                Py_SET_SIZE(z, Py_SIZE(z) + 1);
+                assert(!_PyLong_IsNegative(z));
+                _PyLong_SetSignAndDigitCount(z, 1, _PyLong_DigitCount(z) + 1);
```

Comment:
Seems a good place to use `_PyLong_UnsignedDigitCount`?
```suggestion
                _PyLong_SetSignAndDigitCount(z, 1, _PyLong_UnsignedDigitCount(z) + 1);
```
Alternatively, the code strongly implies z is never negative.

Reply:
I've removed `_PyLong_UnsignedDigitCount`.